advanced_server.rb

```diff
@@ -0,0 +1,319 @@
+          take_requested_action
+        end
+      end
+    end
+    status 200
+  end
+
+
+  helpers do
+
+    # Create a new check run
+    def create_check_run
+      # Octokit doesn't yet support the checks API, but it does provide generic HTTP methods we can use!
+      # https://developer.github.com/v3/checks/runs/#create-a-check-run
+      check_run = @installation_client.post("repos/#{@payload['repository']['full_name']}/check-runs", {
+          accept: 'application/vnd.github.antiope-preview+json', # This header allows for beta access to Checks API
+          name: 'Octo Rubocop',
+          # The information we need should probably be pulled from persistent storage, but we can
+          # use the event that triggered the run creation. However, the structure differs depending on whether
+          # it was a check run or a check suite event that trigged this call.
+          head_sha: @payload['check_run'].nil? ? @payload['check_suite']['head_sha'] : @payload['check_run']['head_sha']
+      })
+
+      # We've now requested the creation of a check run from GitHub. We will wait until we get a confirmation
+      # from GitHub, and then kick off our CI process from there.
+    end
+
+    # Start the CI process
+    def initiate_check_run
+      # This method is called in response to GitHub acknowledging our request to create a check run.
+      # We'll first update the check run to "in progress"
+      # Then we'll run our CI process
+      # Then we'll update the check run to "completed" with the CI results.
+
+      # Octokit doesn't yet support the Checks API, but it does provide generic HTTP methods we can use!
+      # https://developer.github.com/v3/checks/runs/#update-a-check-run
+      # notice the verb! PATCH!
+      updated_check_run = @installation_client.patch("repos/#{@payload['repository']['full_name']}/check-runs/#{@payload['check_run']['id']}", {
+          accept: 'application/vnd.github.antiope-preview+json', # This header is necessary for beta access to Checks API
+          name: 'Octo Rubocop',
+          status: 'in_progress',
+          started_at: Time.now.utc.iso8601
+      })
+
+      # ***** RUN A CI TEST *****
+      # This is where we would kick off our CI process. Ideally this would be performed async, so we could
+      # return immediately. But for now we'll do a simulated CI process syncronously, and update the check run right here..
+
+      full_repo_name = @payload['repository']['full_name']
+      repository     = @payload['repository']['name']
+      head_sha       = @payload['check_run']['head_sha']
+      repository_url = @payload['repository']['html_url']
+
+      clone_repository(full_repo_name, repository, head_sha)
+
+      @report = `rubocop '#{repository}/*' --format json`
+      `rm -rf #{repository}`
+      @output = JSON.parse @report
+      annotations = []
+
+      if @output["summary"]["offense_count"] == 0
+        conclusion = 'success'
+      else
+        conclusion = 'neutral'
```

Comment:
I'm curious why "neutral" and not "failure"?

Reply:
@DEGoodmanWilson that's a good question, and maybe we should change it to "failure." I did this for two reasons:
1. To demonstrate that a "neutral" status could be used (which is not a good demonstration unless we explain this in the quickstart - noted 😄)
2. I chose "neutral" because linters can have very specific rules that might act as more of guidelines rather than rigid rules. However, because the fix button only allows fixing all of the errors, it might be better to use "failure" here.